### PR TITLE
acmecape: better enforce termination

### PR DESCRIPTION
**Pull request by derkling**
This should improve a bit the termination of iio-caputure, thus releasing the channel.

### CODE REVIEW

File: devlib/instrument/acmecape.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 from fcntl import fcntl, F_GETFL, F_SETFL
 from string import Template
 from subprocess import Popen, PIPE, STDOUT
+from signal import SIGTERM
```
Comment (setrofim) [inline devlib/instrument/acmecape.py]: This is imported but never used.

File: devlib/instrument/acmecape.py [hunk 2]
```diff
@@ -81,27 +82,41 @@ def start(self):
...
             if self.process.poll() is not None:
+                still_running = False
                 break
-            time.sleep(1)
         else:
             output += _read_nonblock(self.process.stdout)
-            self.process.kill()
-            self.logger.error('iio-capture did not terminate gracefully')
-            if self.process.poll() is None:
+            # Kill a non responing process (SIGKILL)
```
Comment (marcbonnici) [inline devlib/instrument/acmecape.py]: "responing" -> "responding"

File: devlib/instrument/acmecape.py [hunk 3]
```diff
@@ -81,27 +82,41 @@ def start(self):
...
+                    still_running = False
+                    break
+            else:
+                self.logger.error('iio-capture did not terminate gracefully')
                 msg = 'Could not terminate iio-capture:\n{}'
                 raise HostError(msg.format(output))
-        if self.process.returncode != 15: # iio-capture exits with 15 when killed
+
+        # iio-capture returns the signal sent to stop it
+        if still_running:
```
Comment (marcbonnici) [inline devlib/instrument/acmecape.py]: Can this condition ever be reached with the value being `True`?
Also this does not seem be checking for other error codes being returned anymore if the process has terminated?

**Comment by derkling** [comment]
This looks outdated and I don't see this issue anymore so... let's close it! ;)

**Pull request closed without merging**